sim/mage/arcane/arcane.go

```diff
@@ -34,16 +40,67 @@ func NewArcaneMage(character *core.Character, options *proto.Player) *ArcaneMage
 	return arcaneMage
 }
 
-type ArcaneMage struct {
-	*mage.Mage
-
-	Options *proto.ArcaneMage_Options
-}
-
 func (arcaneMage *ArcaneMage) GetMage() *mage.Mage {
 	return arcaneMage.Mage
 }
 
 func (arcaneMage *ArcaneMage) Reset(sim *core.Simulation) {
 	arcaneMage.Mage.Reset(sim)
 }
+
+func (arcaneMage *ArcaneMage) Initialize() {
+	arcaneMage.Mage.Initialize()
+
+	arcaneMage.registerArcaneBarrageSpell()
+}
+
+func (arcaneMage *ArcaneMage) ApplyTalents() {
+
+	arcaneMage.Mage.ApplyTalents()
+	// Arcane Specialization Bonus
+	arcaneMage.AddStaticMod(core.SpellModConfig{
+		School:     core.SpellSchoolArcane,
+		FloatValue: 0.25,
+		Kind:       core.SpellMod_DamageDone_Flat,
+	})
+
+	// Arcane Mastery
+
+	arcaneMastery := arcaneMage.AddDynamicMod(core.SpellModConfig{
+		ClassMask: mage.MageSpellsAll,
+		Kind:      core.SpellMod_DamageDone_Pct,
+	})
+
+	arcaneMage.AddOnMasteryStatChanged(func(sim *core.Simulation, oldMastery, newMastery float64) {
+		arcaneMastery.UpdateFloatValue(arcaneMage.ArcaneMasteryValue())
+	})
+
+	core.MakePermanent(arcaneMage.GetOrRegisterAura(core.Aura{
+		Label: "Mana Adept",
+		//ActionID: core.ActionID{SpellID: 76547},
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			arcaneMastery.UpdateFloatValue(arcaneMage.ArcaneMasteryValue())
+			arcaneMastery.Activate()
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			arcaneMastery.Deactivate()
+		},
+	}))
+
+	core.MakeProcTriggerAura(&arcaneMage.Unit, core.ProcTrigger{
+		Name:     "Arcane Mastery Mana Updater",
+		Callback: core.CallbackOnCastComplete,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			arcaneMastery.UpdateFloatValue(arcaneMage.ArcaneMasteryValue())
+		},
+	})
+
+}
+
+func (arcaneMage *ArcaneMage) GetArcaneMasteryBonus() float64 {
+	return (1.12 + 0.015*arcaneMage.GetMasteryPoints())
```

Comment:
`0.12 +`

Reply:
ah yeah. remnant from me testing if things were getting affected by it